src/lib/pickles/pickles.ml

```diff
@@ -899,33 +928,30 @@ let compile_promise :
          , a_value
          , (max_proofs_verified, max_proofs_verified) Proof.t Promise.t )
          H3_2.T(Prover).t =
- fun ?self ?(cache = []) ?disk_keys (module A_var) (module A_value) ~typ
-     ~branches ~max_proofs_verified ~name ~constraint_constants ~choices ->
+ fun ?self ?(cache = []) ?disk_keys ~typ ~branches ~max_proofs_verified ~name
+     ~constraint_constants ~choices () ->
```

Comment:
It lets you skip any of the optional arguments; otherwise, you have to pass them all every time, which kinda defeats the point!

In general, labelled and optional arguments can be passed in any order, so you need at least one positional argument for the compiler to make use of their optional potential.

Reply:
gotcha, thanks!